### ISSUE TITLE
Unable to load in moveset_base

### ISSUE THREAD

**Issue by AntonioOlivaH** [megalon/nasb-moveset-editor#26]
I am unsure if it's because the structure changed with a new update, or if I'm doing something wrong, but the editor gets the following error when trying to load the base character moveset, while loading a character specific one ("moveset_clay" for example) works just fine. What could be causing this?

```ReadException: Cannot read an AgentState from tid: 21 at reader int position: 21928, float position: 2734, string position: 2820
NASB_Parser.AgentState.Read (NASB_Parser.BulkSerializeReader reader) (at <52ee8e26fd5d44b088a47c342ad7cc60>:0)
NASB_Parser.IdState..ctor (NASB_Parser.BulkSerializeReader reader) (at <52ee8e26fd5d44b088a47c342ad7cc60>:0)
NASB_Parser.SerialMoveset+<>c.<.ctor>b__5_0 (NASB_Parser.BulkSerializeReader r) (at <52ee8e26fd5d44b088a47c342ad7cc60>:0)
NASB_Parser.BulkSerializeReader.ReadList[T] (System.Func`2[T,TResult] initializer) (at <52ee8e26fd5d44b088a47c342ad7cc60>:0)
NASB_Parser.SerialMoveset..ctor (NASB_Parser.BulkSerializeReader reader) (at <52ee8e26fd5d44b088a47c342ad7cc60>:0)
NASB_Moveset_Editor.GraphHandler.LoadTextAsset () (at Assets/Editor/NASB Moveset Editor/Scripts/GraphHandler.cs:26)
NASB_Moveset_Editor.GraphHandlerWindow.OnGUI () (at Assets/Editor/NASB Moveset Editor/Scripts/GraphHandlerWindow.cs:80)
UnityEditor.HostView.InvokeOnGUI (UnityEngine.Rect onGUIPosition) (at <1135c66e5f4c41a7831fa5798849d8b6>:0)
UnityEditor.DockArea.DrawView (UnityEngine.Rect dockAreaRect) (at <1135c66e5f4c41a7831fa5798849d8b6>:0)
UnityEditor.DockArea.OldOnGUI () (at <1135c66e5f4c41a7831fa5798849d8b6>:0)
UnityEngine.UIElements.IMGUIContainer.DoOnGUI (UnityEngine.Event evt, UnityEngine.Matrix4x4 parentTransform, UnityEngine.Rect clippingRect, System.Boolean isComputingLayout, UnityEngine.Rect layoutSize, System.Action onGUIHandler, System.Boolean canAffectFocus) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.IMGUIContainer.HandleIMGUIEvent (UnityEngine.Event e, UnityEngine.Matrix4x4 worldTransform, UnityEngine.Rect clippingRect, System.Action onGUIHandler, System.Boolean canAffectFocus) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.IMGUIContainer.HandleIMGUIEvent (UnityEngine.Event e, System.Action onGUIHandler, System.Boolean canAffectFocus) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.IMGUIContainer.HandleIMGUIEvent (UnityEngine.Event e, System.Boolean canAffectFocus) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.IMGUIContainer.SendEventToIMGUIRaw (UnityEngine.UIElements.EventBase evt, System.Boolean canAffectFocus, System.Boolean verifyBounds) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.IMGUIContainer.SendEventToIMGUI (UnityEngine.UIElements.EventBase evt, System.Boolean canAffectFocus, System.Boolean verifyBounds) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.IMGUIContainer.HandleEvent (UnityEngine.UIElements.EventBase evt) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.CallbackEventHandler.HandleEventAtTargetPhase (UnityEngine.UIElements.EventBase evt) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.MouseCaptureDispatchingStrategy.DispatchEvent (UnityEngine.UIElements.EventBase evt, UnityEngine.UIElements.IPanel panel) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.EventDispatcher.ApplyDispatchingStrategies (UnityEngine.UIElements.EventBase evt, UnityEngine.UIElements.IPanel panel, System.Boolean imguiEventIsInitiallyUsed) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.EventDispatcher.ProcessEvent (UnityEngine.UIElements.EventBase evt, UnityEngine.UIElements.IPanel panel) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.EventDispatcher.ProcessEventQueue () (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.EventDispatcher.OpenGate () (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.EventDispatcherGate.Dispose () (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.EventDispatcher.ProcessEvent (UnityEngine.UIElements.EventBase evt, UnityEngine.UIElements.IPanel panel) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.EventDispatcher.Dispatch (UnityEngine.UIElements.EventBase evt, UnityEngine.UIElements.IPanel panel, UnityEngine.UIElements.DispatchMode dispatchMode) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.BaseVisualElementPanel.SendEvent (UnityEngine.UIElements.EventBase e, UnityEngine.UIElements.DispatchMode dispatchMode) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.UIElementsUtility.DoDispatch (UnityEngine.UIElements.BaseVisualElementPanel panel) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.UIElementsUtility.UnityEngine.UIElements.IUIElementsUtility.ProcessEvent (System.Int32 instanceID, System.IntPtr nativeEventPtr, System.Boolean& eventHandled) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.UIEventRegistration.ProcessEvent (System.Int32 instanceID, System.IntPtr nativeEventPtr) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.UIElements.UIEventRegistration+<>c.<.cctor>b__1_2 (System.Int32 i, System.IntPtr ptr) (at <a1bb4781672a473da2508a63172fb75e>:0)
UnityEngine.GUIUtility.ProcessEvent (System.Int32 instanceID, System.IntPtr nativeEventPtr, System.Boolean& result) (at <5a00c575c9464227a3227baf016d387b>:0)```

**Comment by megalon** [comment]
Yes, this is likely related to the game updating, and the moveset data changing.
I will need to update my fork of NASB_Parser, NASB_Parser_for_XNode, and the add the new nodes into the moveset editor.

I have a WIP pull request over on the main NASB Parser repo, and when that's ready I'll have to bring it into my forked version for use with Unity.
<https://github.com/sc2ad/NASB_Parser/pull/4>

**Comment by Gs-Flint** [comment]
Any update on this? I've been trying to see if I can update the parsers myself but I've gotten no luck

**Comment by megalon** [comment]
Closed as of #27 
Please update to the latest release
